go/obscuronode/enclave/sql/edbattestation.go

```diff
@@ -0,0 +1,204 @@
+package sql
+
+// todo: was hoping to use the edgeless tool ERA as a dependency for verifying the edgeless db attestations but I'm hitting
+//   dependency issues from `eclient` so progressing by duplicating relevant part of ERA using their `enclave` client
+//   library instead of `eclient`
+
+// ERA (Edgeless remote attestation) is a simple protocol for verifying edgeless tools. It's basically just a small json schema
+// that they use as a standard data blob to encrypt into their attestation reports, includes signerID, security version etc.
+
+// The only changes from https://github.com/edgelesssys/era/blob/master/era/era.go is the use of enclave.VerifyRemoteReport
+// in place of eclient.VerifyRemoteReport and a few tweaks to get our lint to pass.
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"crypto/tls"
+	"encoding/binary"
+	"encoding/hex"
+	"encoding/json"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"github.com/edgelesssys/ego/attestation"
+	"github.com/edgelesssys/ego/attestation/tcbstatus"
+	"github.com/edgelesssys/ego/enclave"
+	"github.com/tidwall/gjson"
+)
+
+type certQuoteResp struct {
+	Cert  string
+	Quote []byte
+}
+
+// ErrEmptyQuote defines an error type when no quote was received. This likely occurs when the host is running in OE Simulation mode.
+var ErrEmptyQuote = errors.New("no quote received")
+
+// GetCertificate gets the TLS certificate from the server in PEM format. It performs remote attestation
+// to verify the certificate. A config file must be provided that contains the attestation metadata.
+func GetCertificate(host, configFilename string) ([]*pem.Block, tcbstatus.Status, error) {
+	config, err := ioutil.ReadFile(configFilename)
+	if err != nil {
+		return nil, tcbstatus.Unknown, err
+	}
+	return getCertificate(host, config, enclave.VerifyRemoteReport)
+}
+
+// InsecureGetCertificate gets the TLS certificate from the server in PEM format, but does not perform remote attestation.
+func InsecureGetCertificate(host string) ([]*pem.Block, error) {
```

Comment:
@BedrockSquirrel Can you delete this method? It's not used. Then you can delete the `verifyFunc` type, and just use `enclave.VerifyRemoteReport` directly inside `getCertificate`, instead of passing it as a parameter.